cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/MultivariateAnomalyDetection.scala

```diff
@@ -232,161 +237,107 @@ trait MADBase extends HasOutputCol
+    )
+    true
+  }
+
+  val intermediateSaveDir = new Param[String](
+    this,
+    "intermediateSaveDir",
+    "Blob storage location in HDFS where intermediate data is saved while training.",
+    isValid = validateIntermediateSaveDir _
+  )
 
   def setIntermediateSaveDir(v: String): this.type = set(intermediateSaveDir, v)
 
   def getIntermediateSaveDir: String = $(intermediateSaveDir)
 
-  val connectionString = new Param[String](this, "connectionString", "Connection String " +
-    "for your storage account used for uploading files.")
-
-  def setConnectionString(v: String): this.type = set(connectionString, v)
-
-  def getConnectionString: Option[String] = get(connectionString)
-
-  val storageName = new Param[String](this, "storageName", "Storage Name " +
-    "for your storage account used for uploading files.")
-
-  def setStorageName(v: String): this.type = set(storageName, v)
-
-  def getStorageName: Option[String] = get(storageName)
-
-  val storageKey = new Param[String](this, "storageKey", "Storage Key " +
-    "for your storage account used for uploading files.")
-
-  def setStorageKey(v: String): this.type = set(storageKey, v)
-
-  def getStorageKey: Option[String] = get(storageKey)
-
-  val endpoint = new Param[String](this, "endpoint", "End Point " +
-    "for your storage account used for uploading files.")
-
-  def setEndpoint(v: String): this.type = set(endpoint, v)
-
-  def getEndpoint: Option[String] = get(endpoint)
-
-  val sasToken = new Param[String](this, "sasToken", "SAS Token " +
-    "for your storage account used for uploading files.")
-
-  def setSASToken(v: String): this.type = set(sasToken, v)
-
-  def getSASToken: Option[String] = get(sasToken)
-
-  val containerName = new Param[String](this, "containerName", "Container that will be " +
-    "used to upload files to.")
-
-  def setContainerName(v: String): this.type = set(containerName, v)
-
-  def getContainerName: String = $(containerName)
-
   setDefault(
     outputCol -> (this.uid + "_output"),
     errorCol -> (this.uid + "_error"))
 
-  protected def getBlobContainerClient: BlobContainerClient = {
-    if (this.get(connectionString).nonEmpty) {
-      getBlobContainerClient(getConnectionString.get, getContainerName)
-    } else if (this.get(storageName).nonEmpty && this.get(storageKey).nonEmpty &&
-      this.get(endpoint).nonEmpty && this.get(sasToken).nonEmpty) {
-      getBlobContainerClient(getStorageName.get, getStorageKey.get, getEndpoint.get,
-        getSASToken.get, getContainerName)
-    } else {
-      throw new Exception("You need to set either {connectionString, containerName} or " +
-        "{storageName, storageKey, endpoint, sasToken, containerName} in order to access the blob container")
-    }
-  }
+  private def getStorageInfo: StorageInfo = {
+    val uri = new URI(getIntermediateSaveDir)
+    val account = uri.getHost.split(".".toCharArray).head
+    val config = s"fs.azure.account.key.$account.blob.core.windows.net"
```

Comment:
Does this config also work on databricks?

Reply:
yes!